app/models/concerns/host_aspects/managed_host_extensions.rb

```diff
@@ -0,0 +1,75 @@
+require 'host_aspects'
+
+module HostAspects
+  module ManagedHostExtensions
+    extend ActiveSupport::Concern
+
+    included do
+      HostAspects::ManagedHostExtensions.refresh_aspect_relations(self)
+
+      has_many :host_aspects, :foreign_key => :host_id, :inverse_of => :host do
+        def [](subject)
+          where(:aspect_subject => subject).first ||
+          select { |aspect| aspect.aspect_subject == subject}.first ||
```

Comment:
I need the second part of the `||` statement for the cases when the record is still new. In this case, the first part will return `[]` (as there are no records in the DB that match the condition) and all the actual records will be found in a local array that gets searched by the second part.
Probably it would be nicer to use `#find`, I will change that.